### PR TITLE
AI archive summary

### PR DESCRIPTION
**Pull request by leobz**
Preview:

image
image

### CODE REVIEW

File: lib/bookmark/archives.ex [hunk 1]
```diff
@@ -30,6 +30,10 @@ defmodule Bookmark.Archives do
     Repo.all(query)
   end
 
+  def get_archives_by_user(user) do
```
Comment (joedavison) [inline lib/bookmark/archives.ex:33]: We should add test coverage for this function

File: lib/bookmark_web/controllers/archive_controller.ex [hunk 2]
```diff
@@ -136,7 +129,7 @@ defmodule BookmarkWeb.ArchiveController do
       [_err, id] = String.split(List.first(regex_result), "archive/")
       user = conn.assigns.current_user
 
-      Bookmark.Archives.create_archive(%{name: id, comment: ""}, user)
+      Bookmark.Archives.create_archive(%{name: id, comment: "", title: url}, user)
```
Comment (joedavison) [inline lib/bookmark_web/controllers/archive_controller.ex:132]: It's unclear in this context what "title: url" refers to

Comment (joedavison) [reply lib/bookmark_web/controllers/archive_controller.ex:132]: I see what this is now. We should not use a url as a title though. Why do we need to provide a title before archive is created? We could pull the actual page title from the index.json we get back from archivebox. So this title would match the TITLE tag of the target archive.

File: summarize.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import os
+import openai
+import PyPDF2
+import sys
+import tiktoken
+
+openai.api_key = os.getenv("OPENAI_API_KEY")
+
+
+#####################     Funtions      ###################
```
Comment (joedavison) [inline summarize.py:12]: Typo: Funtions -> Functions

File: lib/bookmark/archives.ex (hunk 1 above)
Comment (leobz) [reply lib/bookmark/archives.ex:33]: Done!

File: lib/bookmark_web/controllers/archive_controller.ex (hunk 2 above)
Comment (leobz) [reply lib/bookmark_web/controllers/archive_controller.ex]: Done!

File: summarize.py (hunk 3 above)
Comment (leobz) [reply summarize.py]: done!

File: lib/bookmark_web/templates/layout/root.html.heex [hunk 4]
```diff
@@ -20,8 +20,13 @@
       src={Routes.static_path(@conn, "/assets/app.js")}
     >
     </script>
+
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-KK94CHFLLe+nY2dmCWGMq91rCGa5gtU4mk92HdvYe+M/SXH301p5ILy+dN9+nJOZ" crossorigin="anonymous">
```
[the remark this answers is not in the document]
Comment (leobz) [reply lib/bookmark_web/templates/layout/root.html.heex:25]: Yes, I created this issue to migrate this part and apply the Tailwilnd template -> https://github.com/bookmark-org/bookmark/issues/52. What do you think?

[the remark this answers is not in the document]
Comment (joedavison) [reply lib/bookmark_web/templates/layout/root.html.heex:25]: I would like to remove this reference before merging

**Comment by joedavison** [comment]
LGTM and ready to merge pending resolution of open comments (remove tailwind)

File: lib/bookmark_web/templates/layout/root.html.heex (hunk 4 above)
[the remark this answers is not in the document]
Comment (leobz) [reply lib/bookmark_web/templates/layout/root.html.heex]: I uploaded new changes!

**Pull request merged**